src/wrapped_radix_engine_toolkit/low_level/execution_summary/execution_summary.rs

```diff
@@ -0,0 +1,164 @@
+        detailed_classification: impl IntoIterator<Item = DetailedManifestClass>,
+        fee_locks: impl Into<FeeLocks>,
+        fee_summary: impl Into<FeeSummary>,
+        new_entities: impl Into<NewEntities>,
+    ) -> Self {
+        Self {
+            addresses_of_account_withdraws: addresses_of_account_withdraws
+                .into(),
+            addresses_of_account_deposits: addresses_of_account_deposits.into(),
+            addresses_of_accounts_requiring_auth:
+                addresses_of_accounts_requiring_auth
+                    .into_iter()
+                    .collect_vec(),
+            addresses_of_identities_requiring_auth:
+                addresses_of_identities_requiring_auth
+                    .into_iter()
+                    .collect_vec(),
+            newly_created_non_fungibles: newly_created_non_fungibles
+                .into_iter()
+                .collect_vec(),
+            reserved_instructions: reserved_instructions
+                .into_iter()
+                .collect_vec(),
+            presented_proofs: presented_proofs.into_iter().collect_vec(),
+            encountered_component_addresses: encountered_component_addresses
+                .into_iter()
+                .collect_vec(),
+            detailed_classification: detailed_classification
+                .into_iter()
+                .collect_vec(),
+            fee_locks: fee_locks.into(),
+            fee_summary: fee_summary.into(),
+            new_entities: new_entities.into(),
+        }
+    }
+}
+
+fn addresses_of_accounts_from_ret(
+    ret: IndexMap<ScryptoComponentAddress, Vec<RetResourceIndicator>>,
+    network_id: NetworkID,
+) -> HashMap<AccountAddress, Vec<ResourceIndicator>> {
+    ret.into_iter()
+        .map(|p| {
+            (
+                AccountAddress::from((p.0, network_id)),
+                p.1.into_iter()
+                    .map(|i| (i, network_id))
+                    .map(ResourceIndicator::from)
+                    .collect_vec(),
+            )
+        })
+        .collect::<HashMap<_, _>>()
+}
+
+impl From<(RetExecutionSummary, NetworkID)> for ExecutionSummary {
+    fn from(value: (RetExecutionSummary, NetworkID)) -> Self {
+        let (ret, n) = value;
+
+        let mut newly_created_non_fungibles =
+            to_vec_network_aware(ret.newly_created_non_fungibles, n);
+        newly_created_non_fungibles.sort();
+
+        Self::new(
+            addresses_of_accounts_from_ret(ret.account_withdraws, n),
+            addresses_of_accounts_from_ret(ret.account_deposits, n),
+            to_vec_network_aware(ret.accounts_requiring_auth, n),
+            to_vec_network_aware(ret.identities_requiring_auth, n),
+            newly_created_non_fungibles,
+            ret.reserved_instructions
+                .into_iter()
+                .map(ReservedInstruction::from),
+            ret
+                // iOS Wallet only use `Vec<ResourceAddress>` for `presented_proofs` today,
```

Comment:
Only values are [used](https://github.com/radixdlt/babylon-wallet-ios/pull/1038/files#diff-3da210c9d6e67d6ad48b92f186f9081165db746479b2570251fd43339ae82b52)

Reply:
yes? your comment confirms my impl? :) this Swift code:

```
exctractProofs(summary.presentedProofs.values.flatMap { $0 }.map(\.resourceAddress))
```

turns a `Map<Key, Vec<ResourceAddress>>` -> `Vec<ResourceAddress>`, which is what I've done in Rust as well :)

Reply:
yes, it confirms your implementation :)